config/roles.yml

```diff
@@ -272,7 +272,7 @@ cross_cluster_search_remote_full_access:
         - 'indices:data/read/search'
 
 # Allow users to operate query assistant
-ml_query_assistant_access:
+query_assistant_access:
   reserved: true
   cluster_permissions:
     - 'cluster:admin/opensearch/ml/execute'
```

Comment:
Can we add this permission here: `cluster:admin/opensearch/ml/config/get`
It is needed to fetch the agent id. Also will the `cluster:admin/opensearch/ml/execute` provide the ability to execute agents?

Reply:
We think the purpose of adding this role allows user-admin to grand an end-user to have permissions to use query assistant, not let end-user to create a new query assistant. We don't think the agent related permission needed.
```
cluster:admin/opensearch/ml/agents/register
cluster:admin/opensearch/ml/agents/get
cluster:admin/opensearch/ml/agents/search
cluster:admin/opensearch/ml/agents/delete
```
@b4sjoo , did you verify with this role, a user can use query assistant?

Reply:
I am also not sure if we need these permissions as there shouldn't be any memory stored for each natural language to query request.
```
cluster:admin/opensearch/ml/memory/conversation/create
cluster:admin/opensearch/ml/memory/interaction/create
```

I believe only the execute agent and get config apis are used for the query assistant feature: https://github.com/opensearch-project/dashboards-observability/blob/2.x/server/routes/query_assist/utils/agents.ts

Reply:
Minimal set was found through e2e test on test os domain:
```
    "cluster:admin/opensearch/ml/execute",
    "cluster:admin/opensearch/ml/memory/conversation/create",
    "cluster:admin/opensearch/ml/memory/interaction/create",
    "cluster:admin/opensearch/ml/memory/interaction/update",
    "cluster:admin/opensearch/ml/predict",
    "cluster:monitor/state",
    "cluster:monitor/health"
```